### PR TITLE
Add travis script

### PR DESCRIPTION
**Pull request by SchSeba**
add .travis yaml file for unit tests and docker push on merge

### CODE REVIEW

**Comment by SchSeba** [comment]
@phoracek please take a look on this PR

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
+language: go
+sudo: required
+dist: trusty
+go: "1.10"
+
+services:
+- docker
+
+env:
+- BUILDER_NAME="ovs-cni/${TRAVIS_JOB_ID}builder"
```
Comment (phoracek) [inline .travis.yml]: I don't see this used anywhere, do we need it?

File: config/release/kustomization.yaml [hunk 2]
```diff
@@ -3,5 +3,5 @@ bases:
 
 patchesStrategicMerge:
 - manager_image_patch.yaml
-apiVersion: v1beta1
+apiVersion: kustomize.config.k8s.io/v1beta1
```
Comment (phoracek) [inline config/release/kustomization.yaml]: could you please comment on this change in the commit message? or preferably move it to a separate commit?

File: config/default/kustomization.yaml [hunk 3]
```diff
@@ -25,9 +25,9 @@ namePrefix: kubemacpool-
   # Only one of manager_auth_proxy_patch.yaml and
   # manager_prometheus_metrics_patch.yaml should be enabled.
 resources:
-- ../rbac/rbac_role.yaml
-- ../rbac/rbac_role_binding.yaml
-- ../manager/manager.yaml
+- rbac/rbac_role.yaml
```
Comment (phoracek) [inline config/default/kustomization.yaml]: could you please comment on this change in the commit message? or preferably move it to a separate commit?

File: cluster/sync.sh [hunk 4]
```diff
@@ -27,7 +27,8 @@ registry=localhost:$registry_port
 
 REGISTRY=$registry make docker-build
 REGISTRY=$registry make docker-push
-make generate-test
+make docker-generate
+make docker-test
```
Comment (phoracek) [inline cluster/sync.sh]: why do you run test during the sync?

File: Makefile [hunk 5]
```diff
@@ -26,7 +26,8 @@ generate-test: manifests
 
 # Generate manifests e.g. CRD, RBAC etc.
 manifests:
-	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go all
+	go run vendor/sigs.k8s.io/controller-tools/cmd/controller-gen/main.go crd --output-dir config/default/crd
```
Comment (phoracek) [inline Makefile]: could you please comment on this change in the commit message? or preferably move it to a separate commit?

**Comment by SchSeba** [comment]
done @phoracek can you make another review?
